tests/integration/s3-offload/src/test/java/org/apache/pulsar/tests/integration/TestS3Offload.java

```diff
@@ -33,58 +31,69 @@
-import org.jboss.arquillian.test.api.ArquillianResource;
-import org.jboss.arquillian.testng.Arquillian;
+import org.apache.pulsar.tests.containers.S3Container;
+import org.apache.pulsar.tests.topologies.PulsarClusterSpec;
+import org.apache.pulsar.tests.topologies.PulsarClusterTestBase;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import org.testng.Assert;
 import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
-public class TestS3Offload extends Arquillian {
+import static java.util.stream.Collectors.joining;
+
+public class TestS3Offload extends PulsarClusterTestBase {
     private static final Logger log = LoggerFactory.getLogger(TestS3Offload.class);
 
     private static final String CLUSTER_NAME = "test";
     private static final int ENTRY_SIZE = 1024;
     private static final int ENTRIES_PER_LEDGER = 1024;
 
-    @ArquillianResource
-    DockerClient docker;
+    @BeforeClass
+    public void setupCluster() throws Exception {
+        PulsarClusterSpec spec = PulsarClusterSpec.builder()
+            .numBookies(2)
+            .numBrokers(1)
+            .clusterName(Stream.of(this.getClass().getSimpleName(), randomName(5))
+                .filter(s -> s != null && !s.isEmpty())
+                .collect(joining("-")))
+            .build();
+
+        super.setupCluster(spec);
+        pulsarCluster.startS3Container();
+    }
+
 
     @BeforeMethod
     public void configureAndStartBrokers() throws Exception {
 
-        String s3ip = DockerUtils.cubeIdsWithLabels(
-                docker, ImmutableMap.of("service", "s3", "cluster", CLUSTER_NAME))
-            .stream().map((c) -> DockerUtils.getContainerIP(docker, c)).findFirst().get();
+        final String s3Hostname = S3Container.NAME;
 
         String brokerConfFile = "/pulsar/conf/broker.conf";
-        for (String b : PulsarClusterUtils.brokerSet(docker, CLUSTER_NAME)) {
-            PulsarClusterUtils.updateConf(docker, b, brokerConfFile,
-                    "managedLedgerMaxEntriesPerLedger", String.valueOf(ENTRIES_PER_LEDGER));
-            PulsarClusterUtils.updateConf(docker, b, brokerConfFile,
-                    "managedLedgerMinLedgerRolloverTimeMinutes", "0");
-            PulsarClusterUtils.updateConf(docker, b, brokerConfFile,
-                    "managedLedgerOffloadDriver", "s3");
-            PulsarClusterUtils.updateConf(docker, b, brokerConfFile,
-                    "s3ManagedLedgerOffloadBucket", "pulsar-integtest");
-            PulsarClusterUtils.updateConf(docker, b, brokerConfFile,
-                    "s3ManagedLedgerOffloadServiceEndpoint", "http://" + s3ip + ":9090");
-        }
 
-        Assert.assertTrue(PulsarClusterUtils.startAllBrokers(docker, CLUSTER_NAME));
-        Assert.assertTrue(PulsarClusterUtils.startAllProxies(docker, CLUSTER_NAME));
+        pulsarCluster.updateBrokerConf(brokerConfFile,
```

Comment:
these settings can be supplied as environment variables when building broker container in PulsarCluster, because s3Hostname is fixed. there is no need to do the tricks of start containers, update conf, start brokers and proxies.